torch/_decomp/decompositions.py

```diff
@@ -1963,6 +1963,11 @@ def uniform(
     )
 
 
+@register_decomposition(aten.exponential_)
+def exponential_(self, rate=1, generator=None):
+    return self.copy_(-1/rate * torch.log(1 - torch.rand_like(self)))
```

Comment:
don't you have to use `generator` in the `torch.rand_like` call if it is not `None`?

Reply:
Thanks @soumith, looks like prim does not support `generator` yet, I see there is a TODO comment https://github.com/pytorch/pytorch/blob/master/torch/_prims/__init__.py#L2669-L2670. 

For now, I think we can just add `assert generator is None`.

Reply:
makes sense.